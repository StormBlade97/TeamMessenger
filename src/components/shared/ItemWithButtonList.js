import React from 'react'

const ItemWithButtonList = props => {

    const {itemList, onItemButtonClick, buttonText} = props

    return (
        <ul>
            {itemList.map(item =>
                <li key={item.id}>
                    {item.name}
                    <button onClick={() => {
                        onItemButtonClick(item.id)
                    }}>{buttonText}</button>
                </li>
            )}
        </ul>
    )
}

export default ItemWithButtonList