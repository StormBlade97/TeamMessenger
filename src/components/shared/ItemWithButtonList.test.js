import React from 'react'
import ItemWithButtonList from './ItemWithButtonList'
import renderer from 'react-test-renderer'

describe('Component: ListItemWithButton', () => {

    it('renders a snapshot', () => {

        const itemList = [
            {id: 1, name: 'ok'},
            {id: 2, name: 'pvp'},
            {id: 3, name: 'lol'},
        ]

        const tree = renderer.create(
            <ItemWithButtonList
                itemList={itemList}
                buttonText="ok"
            />
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })
})