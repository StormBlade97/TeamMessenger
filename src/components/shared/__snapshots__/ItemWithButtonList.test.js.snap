exports[`Component: ListItemWithButton renders a snapshot 1`] = `
<ul>
  <li>
    ok
    <button
      onClick={[Function]}>
      ok
    </button>
  </li>
  <li>
    pvp
    <button
      onClick={[Function]}>
      ok
    </button>
  </li>
  <li>
    lol
    <button
      onClick={[Function]}>
      ok
    </button>
  </li>
</ul>
`;
